django/cantusdb_project/main_app/views/site_stats.py

```diff
@@ -0,0 +1,102 @@
+"""
+A module containing views for pages that provide general site statistics.
+"""
+
+from main_app.models import (
+    Chant,
+    Sequence,
+    Source,
+    Feast,
+    Office,
+    Provenance,
+    Genre,
+    Notation,
+    Century,
+)
+
+from django.contrib.auth.decorators import login_required, user_passes_test
+from django.core.paginator import Paginator
+from django.shortcuts import render
+from django.core.exceptions import PermissionDenied
+
+
+@login_required
+def items_count(request):
+    """
+    Function-based view for the ``items count`` page, accessed with ``content-statistics``
+
+    Update 2022-01-05:
+    This page has been changed on the original Cantus. It is now in the private domain
+
+    Args:
+        request (request): The request
+
+    Returns:
+        HttpResponse: Render the page
+    """
+    # in items count, the number on old cantus shows the total count of a type of object (chant, seq)
+    # no matter published or not
+    # but for the count of sources, it only shows the count of published sources
+    chant_count = Chant.objects.count()
+    sequence_count = Sequence.objects.count()
+    source_count = Source.objects.filter(published=True).count()
+
+    context = {
+        "chant_count": chant_count,
+        "sequence_count": sequence_count,
+        "source_count": source_count,
+    }
+    return render(request, "items_count.html", context)
+
+
+def project_manager_check(user):
+    """
+    A callback function that will be called by the user_passes_test decorator of content_overview.
+
+    Takes in a logged-in user as an argument.
+    Returns True if they are in a "project manager" group, raises PermissionDenied otherwise.
+    """
+    if user.groups.filter(name="project manager").exists():
+        return True
+    raise PermissionDenied
```

Comment:
I feel like this function would make more sense in either `auth.py` or `permissions.py`. Maybe `permissions.py` would be better since this function isn't actually a view.

Reply:
Ok, never mind about other refactoring...I think it is good as is for now.